### PR TITLE
Google Cloud Run tutorial

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes https://github.com/cockroachdb/docs/issues/11020.

I'll update the existing deployment steps in https://www.cockroachlabs.com/docs/stable/movr-flask-deployment.html#global-application-deployment to use private IPs in a separate PR.

### CODE REVIEW

**Review by awoods187 — APPROVED** [review]
> Reviewed 1 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11063)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling)

___
*[v21.1/deploy-app-gcr.md, line 3 at r2](https://reviewable.io/reviews/cockroachdb/docs/11063#-MhTsov1490mDRvcRinW:-MhTsov21VRu4_-L7DCa:b-ce38kx) ([raw file](https://github.com/cockroachdb/docs/blob/4f8fbd8d15c0226d7bfd774c7d510c324b0ccc99/v21.1/deploy-app-gcr.md#L3)):*
> ```Markdown
> ---
> title: Deploy a Containerized Django Application with Google Cloud Run
> summary: Learn how to use Google Cloud Run to deploy a Django application that communicates with CockroachCloud Free Tier.
> ```

This example is awesome. I think the title undersells this a bit because we show to make your entire application serverless (app + db). I know we dont want to get too marketingy in the docs but i think a title like "Learn how to us GCR and CRDB to deploy a completely serverless application" or something like that is valuable.

I dont think Django is as valuable here since, while we are using django, its not the point. The point is GCR + CC. 

What do you think?

**Comment by awoods187** [comment]
@duoclikebook here is an example of the documentation we can add to further SaaS partnerships. cc @jordanlewis

**Review by ericharmeling — COMMENTED** [review]
TFTR @awoods187 !

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11063#-:-MhUE9Lq0cMmwY-aWZZ2:biugl87)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @awoods187)

___
*[v21.1/deploy-app-gcr.md, line 3 at r2](https://reviewable.io/reviews/cockroachdb/docs/11063#-MhTsov1490mDRvcRinW:-MhU09OD7qUDSAv65aHV:bcpdigd) ([raw file](https://github.com/cockroachdb/docs/blob/4f8fbd8d15c0226d7bfd774c7d510c324b0ccc99/v21.1/deploy-app-gcr.md#L3)):*
<details><summary><i>Previously, awoods187 (Andy Woods) wrote…</i></summary><blockquote>

This example is awesome. I think the title undersells this a bit because we show to make your entire application serverless (app + db). I know we dont want to get too marketingy in the docs but i think a title like "Learn how to us GCR and CRDB to deploy a completely serverless application" or something like that is valuable.

I dont think Django is as valuable here since, while we are using django, its not the point. The point is GCR + CC. 

What do you think?
</blockquote></details>

I think you are right about removing "Django" from the title, but I think we should keep the title as succinct as possible. I've updated the summary to line up with your recommendation (which should help with SEO), but kept the title short.

File: v21.1/deploy-app-gcr.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    - Cloud Run Admin API
+    - Secret Manager API
+
+1. In a local terminal, configure the `gcloud` command-line tool to use the new project:
+
+    {{site.data.alerts.callout_info}}
+    `gcloud` is included with the [Google Cloud SDK](https://cloud.google.com/sdk) installation.
+    {{site.data.alerts.end}}
+
+    {% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/deploy-app-gcr.md:195]:
```suggestion
    {% include_cached copy-clipboard.html %}
```

File: v21.1/deploy-app-gcr.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+To authenticate with CockroachCloud, your application must have access to the root certificate that you downloaded from the CockroachCloud console. You can store the certificate as a secret with the GCP Secret Manager service.
+
+From the [Secret Manager](https://console.cloud.google.com/security/secret-manager), create a secret for the CockroachCloud cluster's root certificate.
+
+## Step 8. Containerize the application and push it to the registry
+
+1. Build the Docker image locally:
+
+    {% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/deploy-app-gcr.md:212]:
```suggestion
    {% include_cached copy-clipboard.html %}
```

File: v21.1/deploy-app-gcr.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    ~~~ shell
+    $ docker build -t gcr.io/<gcp_project_id>/crdb-sample:v1 .
+    ~~~
+
+    If there are no errors, the container built successfully.
+
+1. Authenticate Docker with GCP's Container Registry:
+
+    {% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/deploy-app-gcr.md:221]:
```suggestion
    {% include_cached copy-clipboard.html %}
```

File: v21.1/deploy-app-gcr.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+    ~~~ shell
+    $ gcloud auth configure-docker
+    ~~~
+
+1. Push the Docker image to the project's registry.
+
+    {% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/deploy-app-gcr.md:228]:
```suggestion
    {% include_cached copy-clipboard.html %}
```

File: v21.1/deploy-app-gcr.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+    - Mount the secret that you created for the CockroachCloud certificate on the `certs` volume, with a full path ending in the name of the cert (e.g., `certs/root.crt`).
+
+        {{site.data.alerts.callout_info}}
+        You might need to grant your GCP service account permissions to access the secret.
+        {{site.data.alerts.end}}
+    - Set an environment variable named `DATABASE_URL` to the connection string for a gateway node on the CockroachCloud cluster, with the root certificate located in the mounted `certs` volume (e.g., `'postgresql://user:password@free-tier.gcp-us-central1.cockroachlabs.cloud:26257/defaultdb?sslmode=verify-full&sslrootcert=certs/root.crt&options=--cluster%3Dable-cattle-1220'`).
+
+After the revision is deployed, you should be able to send requests to the application from a browser, or using a REST client (e.g., `curl`). For example:
+
+{% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/deploy-app-gcr.md:249]:
```suggestion
{% include_cached copy-clipboard.html %}
```

**Review by ianjevans — COMMENTED** [review]
Looks good. Just some include_cached nits.

**Review by ianjevans — APPROVED** [review]

**Pull request merged**